examples/seismic/acoustic/operators.py

```diff
@@ -126,13 +180,17 @@ def ForwardOperator(model, geometry, space_order=4,
         Defaults to False.
     kernel : str, optional
         Type of discretization, 'OT2' or 'OT4'.
+    factor : int, optional
+        Downsampling factor to save snapshots of the wavefield.
     """
     m = model.m
 
     # Create symbols for forward wavefield, source and receivers
+    save_value = geometry.nt if save and factor is None else None
```

Comment:
To make the code more homogeneous and easier to read, what we could do, imho, is the following:

* drop `save` from this TimeFunction (which is just legacy behaviour...)
* systematically create `usnaps` and the corresponding equation
* with an `inf` (aka `sys.maxint` ?) `factor`, no snapshots will be saved at runtime
* Tweak [here](https://github.com/devitocodes/devito/blob/master/devito/types/dense.py#L127) adding a line `if self.size == 0: return`, so that Devito avoids allocating memory entirely if pointless

In my opinion, this will dramatically clean up the code here. 

At the moment, the proliferation of `if factor` logic is still affecting maintainability too much

@mloubout is on vacation ATM, but would be useful to hear from his thoughts about this matter